client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -322,10 +318,11 @@ class SimplePaymentsEdit extends Component {
 
 	render() {
 		const { fieldEmailError, fieldPriceError, fieldTitleError } = this.state;
-		const { attributes, isSelected, isLoadingInitial, instanceId } = this.props;
-		const { content, currency, email, multiple, price, title } = attributes;
+		const { attributes, instanceId, isSelected, simplePayment } = this.props;
+		const { content, currency, email, multiple, paymentId, price, title } = attributes;
 
-		if ( ! isSelected && isLoadingInitial ) {
+		const isLoading = paymentId && ! simplePayment;
```

Comment:
Will this cause us "loading" forever if we do have the ID but API doesn't return `simplePayment` for us?

Reply:
That's a case we need to handle and I'm concerned about. There are a few cases:

- The API returns something we don't understand or don't think is valid. Give an option to start fresh?
- The API doesn't respond for some reason. If we don't have data to update, what do we do?

CC: @thomasguillot

Reply:
https://github.com/Automattic/wp-calypso/pull/28588 has the workings of actually reading the loading state and would serve as a basis for some changes.

Reply:
Right, I remember now that we discussed it. I opened an issue to keep track: https://github.com/Automattic/wp-calypso/issues/28634